### PR TITLE
Implement Sqlite support for the CLI.

### PR DESCRIPTION
**Pull request by mcasper**
Because `Connection` isn't object safe (and probably shouldn't be),
switching based on the desired backend had to be achieved through
branching, handled with a macro wherever possible. This will obviously
grow more and more burdensome as more backends are added, and a better
solution will have to be figured out in the future.

I also split out database related functions into their own module, as it
was starting to feel like too much to juggle in one file.

The tests in the `database` mod each use their own database, as for many
of them we're dropping and creating the database, which can't be wrapped
in a transaction. Using a separate database for each allows us to still
run the tests multi threaded, but does add the extra cost of some setup
and teardown in the code.

The Sqlite portions don't use the `tempdir` crate for generating their
database path because I couldn't figure out an ergonomic way of keeping
the tests able to run for both backends, while keeping the TempDir in
scope (returning a path from a function causes the TempDir to go out of
scope, which removes the directory in its destructor).

### CODE REVIEW

**Comment by sgrif** [comment]
The tests for this fail to compile.

File: bin/test [hunk 1]
```diff
@@ -2,7 +2,8 @@
 set -e
 
 (cd diesel && cargo test --features "unstable chrono sqlite")
-(cd diesel_cli && cargo test)
+(cd diesel_cli && cargo test --features "postgres" --no-default-features)
+(cd diesel_cli && cargo test --features "sqlite" --no-default-features)
```
Comment (sgrif) [inline bin/test]: We need to tell travis to run both of these, too.

File: diesel_cli/Cargo.toml [hunk 2]
```diff
@@ -15,8 +15,12 @@ name = "diesel"
 [dependencies]
 chrono = "^0.2.17"
 clap = "^1.5.5"
-diesel = "^0.4.0"
+diesel = { path = "../diesel", features = ["postgres", "sqlite"] }
```
Comment (sgrif) [inline diesel_cli/Cargo.toml]: This should just be `default-features = false`

**Comment by sgrif** [comment]
Can you break out the stuff that moved into `database.rs` without changing in a separate commit to make it easier to review?

**Comment by mcasper** [comment]
Yeah sorry about that, the first commit should be a little easier to diff now

**Comment by sgrif** [comment]
One comment. Please squash before merging.

**Pull request merged**